selfdrive/thermald/power_monitoring.py

```diff
@@ -157,10 +193,42 @@ def _perform_integration(self, t, current_power):
           if power_used < 0:
             raise ValueError(f"Negative power used! Integration time: {integration_time_h} h Current Power: {power_used} uWh")
           self.power_used_uWh += power_used
+          self.car_battery_power_uWh -= power_used
           self.last_measurement_time = t
       except Exception:
         cloudlog.exception("Integration failed")
 
   # Get the power usage
   def get_power_used(self):
     return int(self.power_used_uWh)
+
+  # See if we need to disable charging
+  def should_disable_charging(self, health, offroad_timestamp):
+    if health is None or offroad_timestamp is None:
+      return False
+
+    now = sec_since_boot()
+    disable_charging = False
+    disable_charging |= (now - offroad_timestamp) > MAX_TIME_OFFROAD_S
+    disable_charging |= (self.car_voltage_mV < (VBATT_PAUSE_CHARGING * 1e3))
+    disable_charging |= (self.car_battery_power_uWh <= 0)
+    disable_charging &= (not health.health.ignitionLine and not health.health.ignitionCan)
+    disable_charging &= (self.params.get("DisablePowerDown") != b"1")
+    return disable_charging
```

Comment:
Hmm, why not use logical operators? mixing these bools and bitwise operators doesn't seem great

Reply:
Is there a short way for doing that? Want to avoid should_start = should_start and <>

Reply:
I don't think so, usually it's done like this in python
```python
disable_charging = (condition and
                    second_condition and
                    third condition)
```

Reply:
Hmm, that would become this:
```return ((((now - offroad_timestamp) > MAX_TIME_OFFROAD_S) or
            (self.car_voltage_mV < (VBATT_PAUSE_CHARGING * 1e3)) or
            (self.car_battery_power_uWh <= 0)) and 
            (not health.health.ignitionLine and not health.health.ignitionCan) and
            (self.params.get("DisablePowerDown") != b"1"))```
Not sure if that's clearer or worse